src/java/org/apache/cassandra/cql3/restrictions/SimpleRestriction.java

```diff
@@ -225,17 +226,22 @@ public void restrict(RangeSet<ClusteringElements> rangeSet, QueryOptions options
 
     private List<ClusteringElements> bindAndGetClusteringElements(QueryOptions options)
     {
+        List<ClusteringElements> elements = new ArrayList<>();
```

Comment:
@samueldlightfoot if you think about that ... if columnsExpression.kind() goes to default case which throws, then you have created an object of an empty list unnecessarily. I think what you could do is to make that list directly in either TOKEN or MULTI_COLUMN case. That way we would save an object creation when it goes to default. 

It would be also interesting to investigate if it is possible (and how frequent) that bindAndGet(options) is empty. Then I would probably check if it is empty and only if it is not I would create that list.

Reply:
A good point. I had assumed the exception creation cost would dwarf list creation. I have tidied this up.

Interesting idea on the empty element checks -- I'd have to do some digging on this.